library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
> We would need to convert a vm_obj that represents a value of type alpha into a pre-term that when elaborated has type alpha. Recall that a lot of information is lost at run time.

The `has_quote` approach should be workable, but I'm still interested in the general problem: If we have a `vm_obj` and a closed elaborated expr representing its type (as should be the case with `parse`), can we not walk both of them in parallel to construct an equivalent expr of the tree of constructors (+ some special-case macros)?